### PR TITLE
TfLite Slice missing datatype support

### PR DESCRIPTION
**Pull request by swatheesh-mcw**
-Adds bf16, f16 support for slice
-Adds bf16, f16 slice unit tests

### CODE REVIEW

File: tensorflow/lite/kernels/slice.cc [hunk 1]
```diff
@@ -254,6 +254,12 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
     case kTfLiteString:
       TF_LITE_SLICE(string);
       break;
+    case kTfLiteFloat16:
+      TF_LITE_SLICE(Eigen::half);
```
Comment (rascani) [inline tensorflow/lite/kernels/slice.cc:258]: Please add include for Eigen/core

File: tensorflow/lite/kernels/slice_test.cc [hunk 2]
```diff
@@ -12,13 +12,13 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+#include <gtest/gtest.h>
```
Comment (rascani) [inline tensorflow/lite/kernels/slice_test.cc:15]: Please move this back

File: tensorflow/lite/kernels/slice_test.cc [hunk 3]
```diff
@@ -309,6 +309,74 @@ TEST_P(SliceOpTest, SliceBool) {
   EXPECT_THAT(m.GetOutput(), ElementsAreArray({false, true}));
 }
 
+TEST_P(SliceOpTest, SliceFloat16) {
+  SliceOpModel<Eigen::half, int32_t> m({3, 2, 3, 1}, {4}, {1, 0, 0, 0}, {4},
```
Comment (rascani) [inline tensorflow/lite/kernels/slice_test.cc:313]: Please add include for Eigen/core

File: tensorflow/lite/kernels/test_util.h [hunk 4]
```diff
@@ -15,6 +15,8 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
 #define TENSORFLOW_LITE_KERNELS_TEST_UTIL_H_
 
+#include <gmock/gmock.h>
```
Comment (rascani) [inline tensorflow/lite/kernels/test_util.h:18]: Please move this back.

**Review by rascani — CHANGES_REQUESTED** [review]

File: tensorflow/lite/kernels/slice.cc (hunk 1 above)
Comment (swatheesh-mcw) [reply tensorflow/lite/kernels/slice.cc:259]: Addressed the requested change in [#55aee76](https://github.com/MCW-Dev/tensorflow/pull/33/commits/55aee761ece414a6958216f60479cd0358f1b3f3)

File: tensorflow/lite/kernels/slice_test.cc (hunk 2 above)
Comment (swatheesh-mcw) [reply tensorflow/lite/kernels/slice_test.cc]: Addressed the requested change in [#55aee76](https://github.com/MCW-Dev/tensorflow/pull/33/commits/55aee761ece414a6958216f60479cd0358f1b3f3)

File: tensorflow/lite/kernels/slice_test.cc (hunk 3 above)
Comment (swatheesh-mcw) [reply tensorflow/lite/kernels/slice_test.cc:314]: Addressed the requested change in [#55aee76](https://github.com/MCW-Dev/tensorflow/pull/33/commits/55aee761ece414a6958216f60479cd0358f1b3f3)

File: tensorflow/lite/kernels/test_util.h (hunk 4 above)
Comment (swatheesh-mcw) [reply tensorflow/lite/kernels/test_util.h]: Addressed the requested change in [#55aee76](https://github.com/MCW-Dev/tensorflow/pull/33/commits/55aee761ece414a6958216f60479cd0358f1b3f3)

**Review by rascani — APPROVED** [review]

**Pull request merged**